app/[lang]/chapters/[slug]/[lesson]/page.tsx

```diff
@@ -108,23 +108,11 @@ export default function Page({ params }) {
     )
   }
 
-  // If account and progress data have been loaded, but there is no progress obj
-  if (!account || !progress) {
-    return (
-      <>
-        <Head />
-        <div className="flex h-full w-full grow flex-col items-center justify-center">
-          <SignIn lang={params.lang} />
-        </div>
-      </>
-    )
-  }
-
-  const lastUnlockedLessonPath = getLastUnlockedLessonPath(progress!)
+  const lastUnlockedLessonPath = getLastUnlockedLessonPath(progress)
   const currentLessonPath = `/${pathData.pageId}/${pathData.chapterId}/${pathData.lessonId}`
   const isRestrictedFromLesson = isLessonCompleted(
```

Comment:
I believe this should be proceeded by a not operand as we are checking that the user has Not completed the specified lesson to determine if they should be redirected.

In testing i am able to type in lesson paths I expect to be locked without being redirected
``` suggestion
  const isRestrictedFromLesson = !isLessonCompleted(
```

Reply:
Here comes @benalleng with his eagle eyes!! 🦅
Great catch! ✨

Since this change fixes a different bug significantly different from the goal of this PR, and since this change is independent of changes done in this PR, it's best to address this change separately to maintain the atomic nature of PR.

Let me create a separate PR for this change!

Reply:
Okay, so this change does depend on changes in this PR (removing the mandatory signup screen)
Let me include this change with this PR.

Thanks again for catching this! 🙌

Reply:
Addressed in 98ad49d932fc391a37a256fa92acfa24b3856fdf